service-clients-core/src/main/java/com/flipkart/poseidon/serviceclients/ServiceContext.java

```diff
@@ -164,4 +178,21 @@ public static void shutDown() {
         isDebug.remove();
         debugResponses.remove();
     }
+
+    static Map<String, Queue<String>> getCollectedHeaders() {
+        return collectedHeaders.get();
+    }
+
+    public static List<String> getCollectedHeaders(String header) {
+        return new ArrayList<>(getCollectedHeaders().get(header.toLowerCase()));
```

Comment:
Use optional as map might not have header which app is passing here. Passing null to arraylist will result in NPE